### ISSUE TITLE
Dataset.as_array() returns array of wrong size for stitched images

### ISSUE THREAD

**Issue by manuelmaeritz** [micro-manager/NDTiffStorage#119]
Hello,

I faced an unexpected behavior in the function `as_array` when using the `stitched=True` option. The returned `dask` array seems to be in the wrong shape. When converting it to a `numpy` array however the shape assumes the right dimensions.

I have tow-channel images of dimensions 2016x852px, which are arranged in two columns and eight rows (no overlap).

**Experienced behavior:**

```
d = data.as_array(stitched=True, z=0)
print(d)
>>> a=np.asarray(d)
print(a.shape)
>>> dask.array<read_one_image, shape=(2, 2016, 852), dtype=uint16, chunksize=(1, 2016, 852), chunktype=numpy.ndarray>
a=np.asarray(d)
print(a.shape)
>>>(2, 6816, 4032)
```
Note that beside the fact that the dimensions are wrong the width and the height of the images are swapped: In the `dask`-array the columns come before the rows, whereas in the `numpy` array the rows come before the columns.

**Expected behavior:**

```
d = data.as_array(stitched=True, z=0)
print(d)
>>> a=np.asarray(d)
print(a.shape)
>>> dask.array<read_one_image, shape=(2, 6816, 4032), dtype=uint16, chunksize=(1, 6816, 4032), chunktype=numpy.ndarray>
a=np.asarray(d)
print(a.shape)
>>>(2, 6816, 4032)
```

**Fix:**

Changing the following rows
```
483    w = self.image_width if not stitched else self._tile_width
484    h = self.image_height if not stitched else self._tile_height

...

553    chunks += (w, h)
```
to
```
483    w = self.image_width if not stitched else self._tile_width*len(self.axes["column"])
484    h = self.image_height if not stitched else self._tile_height*len(self.axes["row"])

...

553    chunks += (h, w)
```
in `nd_tiff_v2.py` seems to solve this problem.

**Suggestion:**

This way however the chunks consists of the already stitched images. For very large stitched images it might be better, though, to have a single chunk for every acquired image, thus a `dask` array of properties
```dask.array<read_one_image, shape=(2, 6816, 4032), dtype=uint16, chunksize=(1, 852, 2016), chunktype=numpy.ndarray>```
This might be useful for some people.

The bug is related to the already fixed issue reported by ptbrown1729: https://github.com/micro-manager/pycro-manager/issues/278

Otherwise very nice package! Thanks for developing this open source project!

**Comment by henrypinkard** [comment]
Yes, I agree that the chunks should not be a full stitched image, as this is memory-inefficient. I'm pretty sure that this behavior works as intended for NDTiff v3, so this bug is probably confined to data collected with older versions of pycro-manager/micro-manager--is that right? I'd guess that there are some changes to v3 that may need to be backported to the v2 code. Do you have any interest in making a pull request to apply a fix? I'd be happy to guide you through the process

**Comment by ieivanov** [comment]
Correct, this was fixed for v3 datasets in #107, but I don't think I've tested the stitched behavior - do let us know if you run into issues with v3 datasets as well.
